core/contracts/financial-templates/implementation/Liquidatable.sol

```diff
@@ -171,26 +171,49 @@ contract Liquidatable is PricelessPositionManager {
      * @dev This method generates an ID that will uniquely identify liquidation
      * for the sponsor.
      * @param sponsor address to liquidate
-     * @param amountToLiquidate amount of liquidatedCollateral that liquidator wants to liquidate
+     * @param collateralPerToken abort the liquidation if the position's collateral per token exceeds this value
+     * @param maxTokensToLiquidate max number of tokes to liquidate
      * @return uuid of the newly created liquidation
      */
-    function createLiquidation(address sponsor, FixedPoint.Unsigned calldata amountToLiquidate)
-        external
-        fees()
-        onlyPreExpiration()
-        returns (uint uuid)
-    {
-        // Attempt to retrieve Position data for sponsor
+    // TODO: Perhaps pass this ID via an event rather than a return value
+    function createLiquidation(
+        address sponsor,
+        FixedPoint.Unsigned calldata collateralPerToken,
+        FixedPoint.Unsigned calldata maxTokensToLiquidate
+    ) external fees() onlyPreExpiration() returns (uint uuid) {
         PositionData storage positionToLiquidate = _getPositionData(sponsor);
-        FixedPoint.Unsigned memory positionCollateral = _getCollateral(positionToLiquidate.rawCollateral);
-        require(positionCollateral.isGreaterThan(0));
 
-        // Caller is required to include in order to prevent front-running attacks that modify the liquidatedCollateral amount
-        // @dev amountToLiquidate is >= underlying collateral because the position is still liquidatable if the underlying collateral declines
-        // while the liquidation transaction is pending
-        require(
-            amountToLiquidate.isGreaterThanOrEqual(positionCollateral.sub(positionToLiquidate.withdrawalRequestAmount))
+        // tokensToLiquidate = minimum of maxTokensToLiquidate and how many tokens are outstanding.
+        FixedPoint.Unsigned memory tokensToLiquidate = FixedPoint.Unsigned(maxTokensToLiquidate.rawValue).isGreaterThan(
```

Comment:
Do you have to pass this to `FixedPoint.Unsigned` like this as this is already a `FixedPoint.Unsigned` from the function param?

Reply:
I checked out your branch and tried without this and it didnt work unless the `rawValue` was first cast to a `FixedPoint.Unsigned`. Curious.

Reply:
Yeah this whole line is weird, open to any better/different ways of doing it.

Reply:
Ugh, `calldata` creates so many problems. One way to do this in a way that's potentially clearer is to implement a `min` function in the FixedPoint library that just piggybacks off of the openzeppelin min function: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e2813df879a4e4e74538d3c8bde0b50af8952597/contracts/math/Math.sol#L17-L19.

So effectively this line could be reduced to:
```
FixedPoint.min(maxTokensToLiquidate, positionToLiquidate.tokensOutstanding)
```

I think this would clear up all the calldata casting issues as an intended side-effect.

Reply:
I like that approach way more, Done.